impeller/typographer/text_frame.cc

```diff
@@ -79,13 +79,23 @@ bool TextFrame::MaybeHasOverlapping() const {
   return false;
 }
 
+// static
+Scalar TextFrame::RoundScaledFontSize(Scalar scale, Scalar point_size) {
+  if (point_size <= 25) {
+    return std::round(scale * 10) / 10;
+  }
+  return std::round(scale * 100) / 100;
+}
```

Comment:
I don't think this covers all cases, would something like this work?

```cpp
// The point_size is multiplied into the total scale of the font later,
// and so we need to linearly increase the step precision as the point size increases.
auto precision = 20 * point_size; // The constant here may need to be adjusted for best perf.
return std::pow(kE, std::round(std::log(scale) * precision) / precision);
```

![image](https://github.com/flutter/engine/assets/919017/1a9f7028-2b9d-41c4-9a8e-afa2ee09915f)

https://www.desmos.com/calculator/sqpcws9bfv

Reply:
This curve is going to introduce a lot of churn for values that are already fairly stable.

For the default text scale of 3.5x (the DPR) we can end up converting this to  3.49908 or  3.50127 based on the exact text scale. I don't we're going to be able to use an log based rounding.